app/models/middleware_server_performance.rb

```diff
@@ -16,7 +16,11 @@ def self.build_results_for_report_middleware(options)
     start_time, end_time, interval = parse_time_interval(options)
     results = []
     MiddlewareServer.find_each do |ms|
-      raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
+      begin
+        raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
```

Comment:
Based on [this](https://github.com/hawkular/hawkular-client-ruby/pull/115#discussion-diff-71081377R185) , there could be many more connection issues, so I am not sure I understand the difference in handling it here and in the gem. Can you please explain? thanks